asab/web/authn/oauth/middleware.py

```diff
@@ -76,13 +62,12 @@ def oauthclient_middleware_factory(app, *args, methods, identity_cache_longevity
 					if oauth_user_info is not None:
 						request.OAuthUserInfo = oauth_user_info
 						request.Identity = method.extract_identity(oauth_user_info)
-						identity_cache[oauth_server_id_access_token] = {
-							"OAuthUserInfo": request.OAuthUserInfo,
-							"Identity": request.Identity,
-							"Expiration": time.time() + identity_cache_longevity
-						}
+						oauth_client_service.IdentityCache[oauth_server_id_access_token] = (request.OAuthUserInfo, request.Identity)
 				else:
-					raise RuntimeError("Call to OAuth server '{}' failed with status code '{}'.".format(oauth_userinfo_url, resp.status))
+					# "authn_required_handler" decorator will then return "HTTPUnauthorized" to the client,
+					# because of missing identity in the request
+					assert not hasattr(request, "Identity")
```

Comment:
Please move this assert to the top of this method ... there are other ways how this could be skipped (and it is still a security concern).